src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3739,6 +3739,15 @@ const char* InstanceKlass::internal_name() const {
 void InstanceKlass::print_class_load_logging(ClassLoaderData* loader_data,
                                              const ModuleEntry* module_entry,
                                              const ClassFileStream* cfs) const {
+  if (TraceClassLoadingCause != nullptr &&
+      (strcmp(TraceClassLoadingCause, "*") == 0 || strstr(external_name(), TraceClassLoadingCause) != nullptr))
```

Comment:
> Not sure if "contains" is the right way to match here ??

What else would you suggest? In my experience, I'm typically interested in a specific class or package so substring matching has been sufficient. I don't think this feature warrants the complexity of regular expression matching.

> Also I don't see how this matches the example you gave for Thread because it only produced a few stacks. ??

Are you commenting on the fact that the first few classes in my example have no stacks? That's because that loading happens in the VM before any Java code is running. Also note that my example does not contain the complete execution trace (as indicated by the `...` at the end).

Reply:
Ah I see, so those initial entries are empty stacks because you are only showing Java frames. I misunderstood. I was thinking a full native trace. If a class is loaded as part of verification of another class, the Java stack might not be that useful.

As for "contains" ... just not sure if it it might not match too many things ... I guess you just have to be more specific about whether you specify a simple class name of FQN.

Reply:
> you are only showing Java frames

I'm not aware of a way to print the native frames (outside of hs-err logging). It would be great if such a mechanism exists.

> if it it might not match too many things

If you make the filter a FQN, you can be very specific.

Reply:
See `JavaThread::print_jni_stack()`

Reply:
Of course we'd need to generalize `print_jni_stack` to take an `outputStream` parameter.